src/InvariantTheory/invariant_rings.jl

```diff
@@ -246,6 +248,93 @@ function reynolds_operator(IR::InvRing, f::MPolyElem)
   return reynolds_operator(IR, polynomial_ring(IR)(f))
 end
 
+function reynolds_operator(IR::InvRing{FldT, GrpT, PolyElemT}, chi::GAPGroupClassFunction) where {FldT, GrpT, PolyElemT}
+# I expect that this also works in the non-modular case, but haven't found a reference.
+  # The only reference for this version of the reynolds operator appears to be [Gat96].
+  @assert is_zero(characteristic(coefficient_ring(IR)))
+  @assert is_irreducible(chi)
+
+  K = coefficient_ring(IR)
+
+  conjchi = conj(chi)
+
+  actions_and_values = [ (right_action(polynomial_ring(IR), g), K(conjchi(g).data)) for g in group(IR) ]
+
+  function reynolds(f::PolyElemT)
+    g = parent(f)()
+    for (action, val) in actions_and_values
+      g = addeq!(g, action(f)*val)
+    end
+    return g*base_ring(f)(1//order(group(IR)))
+  end
+
+  return MapFromFunc(reynolds, polynomial_ring(IR), polynomial_ring(IR))
+end
+
+@doc Markdown.doc"""
+     reynolds_operator(IR::InvRing{FldT, GrpT, T}, f::T, chi::GAPGroupClassFunction)
+       where {FldT, GrpT, T <: MPolyElem}
+
+In the case of characteristic zero, return the image of `f` under the twisted
+Reynolds operator projecting onto the isotypic component of the polynomial ring
+with respect to `chi`, that is, the semi-invariants (or relative invariants)
+with respect to `chi`, see [Sta79](@cite).
+It is assumed that `chi` is an irreducible character.
+
+In case `chi` is a linear character, the returned polynomial, say `h`, fulfils
+`h^g = chi(g)h` for all `g` in `group(IR)` (possibly `h` is zero).
+
+!!! note
+    It is implicitly assumed that `coefficient_ring(IR)` contains all character values of `chi`.
```

Comment:
Is that an expensive check? If it is not too expensive, I would rather add a `check=true` keyword  so that 
the method is safe by default, and if anyone needs full speed an knows what they do, they can disable the check.

Reply:
There is no actual check. I have to cast the character values (which are `QQAbElem`s) in the coefficient field. This either works or there is an error.

Reply:
Okay. Then the behavior is fine to me.  I interpreted 

"It is implicitly assumed that `coefficient_ring(IR)` contains all character values of `chi`."

as: if the implicit assumption is not true, than anything may happen including false result. 
So I would be more concrete and reformulate this to:  
"If the `coefficient_ring(IR)` does not contain all character values of `chi`, an error is raised."

Reply:
Ah, you're right! I changed the wording accordingly. Thank you!